workitem/link/link_repository.go

```diff
@@ -41,6 +42,14 @@ type WorkItemLinkRepository interface {
 	ListWorkItemChildren(ctx context.Context, parentID uuid.UUID, start *int, limit *int) ([]workitem.WorkItem, uint64, error)
 	WorkItemHasChildren(ctx context.Context, parentID uuid.UUID) (bool, error)
 	GetParentID(ctx context.Context, ID uuid.UUID) (*uuid.UUID, error) // GetParentID returns parent ID of the given work item if any
+	// GetAncestors returns all IDs of the ancestors for the given work items.
+	// In addition to that it also returns the root IDs for each given work item
+	// ID.
+	//
+	// NOTE: In case the given link type doesn't have a tree topology a work
+	// item might have more than one root item. That is why the root IDs is
+	// keyed by the the given work item and mapped to an array of root IDs.
+	GetAncestors(ctx context.Context, linkTypeID uuid.UUID, workItemIDs ...uuid.UUID) (distinctAncestorIDs []uuid.UUID, rootIDs map[uuid.UUID][]uuid.UUID, err error)
```

Comment:
@kwk why do you prefer using variable parameter over object array?

Reply:
You can still pass an array to that function (you just need to expand it; see below) and internally it gets treated as an array anyways. I just think it makes things simpler when you only want to lookup one work item. With an array you would have to put it in an array first and then pass it to this function. 

[Here's an example](https://play.golang.org/p/z0kUj5mGbq)

```go
package main
import (
	"fmt"
)

func foo(ints ...int) {
	for idx, i := range ints {
		fmt.Printf("%d: %d\n", idx, i)
	}
}

func main() {
	foo(5, 6, 7)
	foo(10)
	foo([]int{4,5,6}...)
}
```